libminifi/src/utils/HTTPClient.cpp

```diff
@@ -48,72 +73,87 @@ std::string get_token(utils::BaseHTTPClient *client, std::string username, std::
   return token;
 }
 
-void parse_url(const std::string *url, std::string *host, int *port, std::string *protocol) {
-  static std::string http("http://");
-  static std::string https("https://");
-
-  if (url->compare(0, http.size(), http) == 0)
-    *protocol = http;
-
-  if (url->compare(0, https.size(), https) == 0)
-    *protocol = https;
-
-  if (!protocol->empty()) {
-    size_t pos = url->find_first_of(":", protocol->size());
+URL::URL(const std::string& url_input) {
+  if (utils::StringUtils::startsWith(url_input, HTTP)) {
+    protocol_ = HTTP;
+  } else if (utils::StringUtils::startsWith(url_input, HTTPS)) {
+    protocol_ = HTTPS;
+  } else {
+    logger_->log_error("Unknown protocol in URL '%s'", url_input);
+    return;
+  }
```

Comment:
This could be extracted as a separate function `parseProtocol`

Reply:
done in https://github.com/apache/nifi-minifi-cpp/pull/1046/commits/73f95ac1f129ec5fea1fa26710456c86fd88a0c2